src/Razor/src/Microsoft.VisualStudio.Editor.Razor/Telemetry/TelemetryReporter.cs

```diff
@@ -102,11 +104,22 @@ static bool IsComplexValue(object? o)
 
     public void ReportFault(Exception exception, string? message, params object?[] @params)
     {
+        ReportFault(exception, message, 0, @params);
+    }
+
+    private void ReportFault(Exception exception, string? message, int depth, params object?[] @params)
+    {
+        if (depth > RecursionDepthLimit)
+        {
+            // exceeded recursion depth limit
+            return;
+        }
+
         try
         {
             if (exception is OperationCanceledException { InnerException: { } oceInnerException })
             {
-                ReportFault(oceInnerException, message, @params);
+                ReportFault(oceInnerException, message, depth + 1, @params);
```

Comment:
Rather than being recursive, if my previous assumption about `Flatten()` is correct, then this bit of code will over ever process `OperationCanceledException`s that contain other `OperationCanceledException`s, so we could use a very simple method that loops through the `InnerException`s until it gets one that isn't an `OperationCanceledException`, and then call `ReportFault` on that. I think that would completely remove the need for any kind of depth checking.

Reply:
Sure. If the only culprit ends up being OCE I'll just do that. About to confirm with more tests.